core/cat/looking_glass/stray_cat.py

```diff
@@ -576,23 +580,78 @@ def stringify_chat_history(self, latest_n: int = 5) -> str:
 
         history_string = ""
         for turn in history:
-            history_string += f"\n - {turn['who']}: {turn['message']}"
+            history_string += f"\n - {turn.content.who}: {turn.content.text}"
 
         return history_string
 
     def langchainfy_chat_history(self, latest_n: int = 5) -> List[BaseMessage]:
-        chat_history = self.working_memory.history[-latest_n:]
 
+        def format_human_message(message: HumanMessage) -> HumanMessage:
+            """Format a human message, including any text and images."""
+            content = [{"type": "text", "text": message.content.text}]
+            
+            if message.content.images:
+                content.extend(format_images(message.content.images))
+            
+            return HumanMessage(
+                name=message.content.who,
+                content=content
+            )
+
+        def format_ai_message(message) -> AIMessage:
+            """Format an AI message with text content only."""
+            return AIMessage(
+                name=message.content.who,
+                content=message.content.text
```

Comment:
why do you assign only the text to `AIMessage`? Shouldn't the LLM be able to generate images, too (e.g. https://python.langchain.com/docs/integrations/tools/google_imagen/#image-generation)?

Reply:
Because we only have text, this PR is focused on supporting multimodal **LLMs**, imagen is not an LLM but a text-to-image model, as far as I know currently there is no LLM that generates images